docs/source/examples/semantic_segmentation.rst

```diff
@@ -0,0 +1,223 @@
+
+.. code-block:: python
+
+    #callbacks
+    
+    save_path = 'saves'
+    
+    # Creating saving directory 
+    os.makedirs(save_path, exist_ok=True)
+    
+    callbacks = [
+        # Save the latest weights to be able to continue the optimization at the end for more epochs.
+        ModelCheckpoint(os.path.join(save_path, 'last_weights.ckpt')),
+    
+        # Save the weights in a new file when the current model is better than all previous models.
+        ModelCheckpoint(os.path.join(save_path, 'best_weight.ckpt'),
+                        save_best_only=True, restore_best=True, verbose=True),
+    
+        # Save the losses for each epoch in a TSV.
+        CSVLogger(os.path.join(save_path, 'log.tsv'), separator='\t'),
+    ]
+
+Training
+========
+
+.. code-block:: python
+
+    # Poutyne Model on GPU
+    model = Model(network, optimizer, criterion, device=device)
+    
+    # Train
+    model.fit_generator(train_dataloader, valid_dataloader, epochs=num_epochs, callbacks=callbacks)
+
+Calculation of the scores and visualization of results
+======================================================
+
+There is one more helpful feature in Poutyne, which makes the evaluation task more easy and straight forward. Usually, computer vision researchers try to evaluate their trained networks on validation/test datasets by obtaining the scores (accuracy or loss usually), ground truths, and computed results simultaneously. The ``evaluate`` methods in Poutyne provides you with the scores but also have made the other two items ready for further analysis and visualization. In the next few blocks of code, you will see some examples.
+
+.. code-block:: python
+
+    loss, predictions, ground_truth = model.evaluate_generator(valid_dataloader, return_pred=True, return_ground_truth=True)
+
+We show some of the segmentation results in the image below (grayscale):
+
+.. code-block:: python
+
+    outputs = torch.tensor(model.predict_on_batch(inputs))
+    output_grid = make_grid(outputs)
+    out = output_grid.numpy().transpose((1, 2, 0))
+    out=np.clip(out, 0, 1)
+    
+    fig = plt.figure(figsize=(10, 10))
+    plt.imshow((out))
+    plt.show()
+
+.. image:: /_static/img/semantic_segmentation/segment_out.png 
+
+Here, we show one of the input samples along with its segmentation ground truth and the produced output.
+
+.. code-block:: python
+
+    sample_number = 14
+    
+    input_sample = inputs[sample_number].numpy().transpose((1, 2, 0)) * imagenet_std + imagenet_mean
+    ground_truth_sample = ground_truth[sample_number][0]
+    output_sample = outputs[sample_number][0].numpy()
+    
+    fig, (ax1, ax2, ax3) = plt.subplots(1,3)
+    ax1.imshow(input_sample)
+    ax1.axis('off')
+    ax1.set_title('input')
+    
+    ax2.imshow(ground_truth_sample)
+    ax2.axis('off')
+    ax2.set_title('GT')
+    
+    ax3.imshow(output_sample)
+    ax3.axis('off')
+    ax3.set_title('output')
+    plt.show()
```

Comment:
In the notebook, there is an extra line return after this cell. Please, remove it. I am not able to comment on the notebook on Github seems it seems it can't handle its size.